### PR TITLE
Weather monika

### PR DESCRIPTION
**Pull request by MonikaMusial**

### CODE REVIEW

File: 3-12-2018/monika/weather/lib/get_weather.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
+require 'http'
+require 'byebug'
+require 'csv'
+
+class WeatherForecast
+  KEY = "sMKVk9lTLY7N9uRMBf3hQ9Ys9ePGlLsB".freeze
+
+  def initialize(location)
+    city_location = HTTP.get("http://dataservice.accuweather.com/locations/v1/PL/search?apikey=#{KEY}&q=#{location}")
```
Comment (ppietrasik) [inline 3-12-2018/monika/weather/lib/get_weather.rb]: Could be extracted into HOST constant http://dataservice.accuweather.com/

File: 3-12-2018/monika/weather/public/styles.css [hunk 2]
```diff
@@ -0,0 +1,32 @@
+body {
```
Comment (GregoryWilusz) [inline 3-12-2018/monika/weather/public/styles.css]: Cool! :-)

File: 3-12-2018/monika/weather/lib/get_weather.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  def min_temperature()
+    @min_temperatures
+  end
+
+  def rain()
+    @rain
+  end
+
+  def download()   
+    csv_string = CSV.generate do |csv|
```
Comment (ppietrasik) [inline 3-12-2018/monika/weather/lib/get_weather.rb]: No need for csv_string

File: 3-12-2018/monika/weather/lib/get_weather.rb [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+require 'byebug'
+require 'csv'
+
+class WeatherForecast
+  KEY = "sMKVk9lTLY7N9uRMBf3hQ9Ys9ePGlLsB".freeze
+
+  def initialize(location)
+    city_location = HTTP.get("http://dataservice.accuweather.com/locations/v1/PL/search?apikey=#{KEY}&q=#{location}")
+    raise ApiError unless city_location.code == 200  
+        p JSON.parse(city_location.body)[0]
```
Comment (ppietrasik) [inline 3-12-2018/monika/weather/lib/get_weather.rb]: All of those prints to console could be deleted

File: 3-12-2018/monika/weather/views/weather.erb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+          Max temperature: <%= @max_temp[item] %>
+          <br>
+          Rain: <%= @rain[item] %>
+        </div>
+        <% end %>
+        <br>
+      </div>
+      <div>
+        <% if params[:location] %>
+          <a href="/download" download><button class="button">Download</button></a>
```
Comment (GregoryWilusz) [inline 3-12-2018/monika/weather/views/weather.erb]: Great. Whole view is very clear and readable.

File: 3-12-2018/monika/weather/spec/app_spec.rb [hunk 6]
```diff
@@ -0,0 +1,43 @@
+ENV['RACK_ENV'] = 'test'
+
+require 'spec_helper'
+require './app'
+require 'rack/test'
+
+RSpec.describe 'Weather application' do
+  include Rack::Test::Methods
+
+  before do
```
Comment (ppietrasik) [inline 3-12-2018/monika/weather/spec/app_spec.rb]: Could be deleted

File: 3-12-2018/monika/weather/views/weather.erb [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  <link rel="stylesheet" type="text/css" href="styles.css">
+  <title>Weather App</title>
+</head>
+
+<body>
+  <div class="main">
+    <h1>Weather Forecast</h1>Weather Forecast
+    <form class="form">
+      <label for="location">City</label>
+      <input name="location" value="<%= params[:location] %>" />
```
Comment (ppietrasik) [inline 3-12-2018/monika/weather/views/weather.erb]: (To discuss) Is it good practice to use params[] directly in view?

File: 3-12-2018/monika/weather/spec/app_spec.rb [hunk 8]
```diff
@@ -0,0 +1,43 @@
+ENV['RACK_ENV'] = 'test'
```
Comment (ppietrasik) [inline 3-12-2018/monika/weather/spec/app_spec.rb]: Probably could be deleted

File: 3-12-2018/monika/weather/app.rb [hunk 9]
```diff
@@ -0,0 +1,27 @@
+require 'sinatra'
+require_relative 'lib/get_weather'
+
+  get '/download' do
```
Comment (GregoryWilusz) [inline 3-12-2018/monika/weather/app.rb]: Nice! You are the only one who handled download using another path :-)

Comment (ppietrasik) [inline 3-12-2018/monika/weather/app.rb]: Good idea with this separate route

File: 3-12-2018/monika/weather/Gemfile [hunk 10]
```diff
@@ -0,0 +1,7 @@
+source 'https://rubygems.org'
+
+gem 'sinatra'
+gem 'rspec'
```
Comment (ppietrasik) [inline 3-12-2018/monika/weather/Gemfile]: Could be grouped into test

**Pull request merged**